### PR TITLE
Create Controller

### PR DESCRIPTION
**Pull request by enocol**
In this mile stone, the followig was achieved:
- A Users controller was created
- Actions or methods for the following was create
      - Method to get all users as the index of the page
      - Method  to get a specific user with and ID
      - Method to get all post for a specific user
      - Methos to get a specific post for a specific user 
- Routes for the following url parameters were created:
     - Routers to get all users as index page
     - Routes to get a particular user
     - Route to get all post for a particular user
     - Route to get a specific post for a specific user.

### CODE REVIEW

File: app/controllers/users_controller.rb
Comment (IngDavidTM) [inline app/controllers/users_controller.rb:1]: - [ ] You performed exceptionally well; however, as part of the project specifications, it is necessary to include a posts_controller with both #index and #show actions in your project :+1:.

File: app/views/users/show_user_post.html.erb
Comment (IngDavidTM) [inline app/views/users/show_user_post.html.erb:1]: - [ ] Well done on crafting the views for users. However, as a project prerequisite, ensure you have matching view files for each controller action, similar to the following:

```ruby
    # app/views/users/index.html.erb
    # app/views/users/show.html.erb

    # app/views/posts/index.html.erb
    # app/views/posts/show.html.erb
```

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi @enocol 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] You've performed well :+1:, but upon attempting to migrate the database, I encountered the following error. Kindly address and resolve this issue, please.

Screenshot from 2024-01-22 14-32-18


- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb [hunk 1]
```diff
@@ -1,14 +1,23 @@
...
-  resources :comments
-  resources :posts
-  resources :users
+  # resources :likes
+  # resources :comments
+  # resources :posts
+  get 'users/:id', to: 'users#show', as: 'user'
+  get 'users/:id/posts', to: 'users#posts', as: :user_posts
+  get 'users/:id/posts/:id', to: 'users#show_user_post', as: 'show_user_post'
+  get '/users', to: 'users#index', as: 'users'
```
Comment (Olamarx) [inline config/routes.rb:8]: - [OPTIONAL] I would recommend you to use fewer code as 
```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

With this, we will mean the same as you've written

File: app/controllers/users_controller.rb
Comment (Olamarx) [reply app/controllers/users_controller.rb:1]: > * [ ]  You performed exceptionally well; however, as part of the project specifications, it is necessary to include a posts_controller with both #index and #show actions in your project 👍.

- [ ] As the previous reviewer noted, you need to add the post_controller too as required in this milestone. The file should have the index and show methods as you have it for the users_controller

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @enocol, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- GitFlow ✅
- Descriptive and professional PR ✅
- Linters are working properly ✅

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by enocol** [comment]
Hi @Olamarx. Thanks for reviewing my project. Just to get more clarification. You mentioned that as requested by the previous reviewer, that its neccessery to have a Post_controller file with a show and index actions. My project actually has these controller and the specified actions as shown in the image below:
image

Could you explain futher if there is something missing in these actions please. Thanks

**Review by Ibizugbe — APPROVED** [review]
Hi @enocol 👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅

- [x] Index and show actions for post controller have been added👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**